checker/tests/nullness/Issue738.java

```diff
@@ -0,0 +1,41 @@
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+// Testcase for #738
+// https://github.com/typetools/checker-framework/issues/738
+// Also, see framework/tests/all-systems/Issue738.java
+public class Issue738 {
+    public static void methodA(int[] is, Object @Nullable [] os, int i) {
+        // NullnessTransfer changes the type of an agrument is assigned to a @NonNull parameter
+        // then the argument is changed to @NonNull, so use a switch statement to prevent this.
+        switch (i) {
+            case 1:
+                methodB(is, os);
+                break;
+            case 2:
+                //:: error: (argument.type.incompatible)
+                methodB2(is, os);
+                break;
+            case 3:
+                //:: error: (type.argument.type.incompatible)
+                methodB3(is, os);
+                break;
+            case 4:
+                //:: error: (type.argument.type.incompatible)
+                methodB4(is, os);
+                break;
+        }
+    }
+
+    private static <T> void methodB(T paramA, T paramB) {
+    }
+
+    private static <T> void methodB2(T paramA, @NonNull T paramB) {
+    }
+
+    private static <@NonNull T extends @NonNull Object> void methodB3(T paramA, T paramB) {
+    }
+
+    private static <T extends @NonNull Cloneable> void methodB4(T paramA, T paramB) {
```

Comment:
`Cloneable` is an interface, so that upper bound implicitly is `Object & Cloneable`, which is an intersection type.
The comment you added clarified the example.